java/execution/impl/src/com/intellij/execution/filters/ProjectAbsoluteFileLocationFilter.kt

```diff
@@ -0,0 +1,63 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.intellij.execution.filters
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.text.LineColumn
+import java.io.File
+
+/**
+ * @author Bo Zhang
+ */
+class ProjectAbsoluteFileLocationFilter(private val myProject: Project) : Filter {
+  companion object {
+    val FILE_ENDING_EXTENSION_PATTERN = "((\\.[\\w-]+)+)[^\\w-.]*".toPattern()
+  }
+
+  override fun applyFilter(line: String, entireLength: Int): Filter.Result? {
+    return myProject.basePath?.takeIf {
```

Comment:
I expect that `myProject.basePath` will always return the same, so it would be good to evaluate this only once.